public/app/plugins/datasource/elasticsearch/components/hooks/useCreatableSelectPersistedBehaviour.test.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Select, InlineField } from '@grafana/ui';
+import { useCreatableSelectPersistedBehaviour } from './useCreatableSelectPersistedBehaviour';
+import userEvent from '@testing-library/user-event';
+
+describe('useCreatableSelectPersistedBehaviour', () => {
+  it('Should make a Select accept custom values', () => {
+    const MyComp = (_: { force?: boolean }) => (
+      <InlineField label="label">
+        <Select
+          inputId="select"
+          {...useCreatableSelectPersistedBehaviour({
+            options: [{ label: 'Option 1', value: 'Option 1' }],
+            onChange() {},
+          })}
+        />
+      </InlineField>
+    );
+
+    const { rerender } = render(<MyComp />);
+
+    const input = screen.getByLabelText('label') as HTMLInputElement;
+    expect(input).toBeInTheDocument();
+
+    // we open the menu
+    userEvent.click(input);
+
+    expect(screen.getByText('Option 1')).toBeInTheDocument();
+
+    // we type in the input 'Option 2', which should prompt an option creation
+    userEvent.type(input, 'Option 2');
+    const creatableOption = screen.getByLabelText('Select option');
+    expect(creatableOption).toHaveTextContent('Create: Option 2');
+
+    // we click on the creatable option to trigger its creation
+    userEvent.click(creatableOption);
+
+    // Forcing a rerender
+    rerender(<MyComp force={true} />);
+
+    // we open the menu again
+    userEvent.click(input);
+    // the created option should be available
+    expect(screen.getByText('Option 2')).toBeInTheDocument();
+  });
+
+  it('Should handle onChange properly', () => {
+    const onChange = jest.fn();
+    const MyComp = () => (
+      <InlineField label="label">
+        <Select
+          inputId="select"
+          {...useCreatableSelectPersistedBehaviour({
+            options: [{ label: 'Option 1', value: 'Option 1' }],
+            onChange,
+          })}
+        />
+      </InlineField>
+    );
+
+    render(<MyComp />);
+
+    const input = screen.getByLabelText('label') as HTMLInputElement;
+    expect(input).toBeInTheDocument();
+
+    // we open the menu
+    userEvent.click(input);
+
+    // default options don't have 1M but 1m
```

Comment:
This is probably a left-over from the DateHistogramSettingsEdittor test that should be deleted.

Reply:
trueee! thanks!